### DISCUSSION TITLE
[Enhancement] [notification] Add `showProgressBar`, `actions` and `timerControls` props

### DISCUSSION THREAD

**Discussion by geforseN** [element-plus/element-plus#18542] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `notification`

## Description

I am transitioning code from Nuxt UI Notification to Element Plus Notification. However, some of the features available in Nuxt UI Notification are not present in `ElNotification`. To close this gap, I propose adding the following features. **If you find these additions useful, I will submit a pull request to implement them.**

### Feature Requests:

1. **Progress Bar for Timer:**  
  Display a progress bar at the bottom of the notification when `timeout` is a positive number. This will provide users with a visual indication of the remaining time before the notification closes.

2. **Action Buttons:**  
  Allow the addition of action buttons to notifications, each with a label and a handler, enabling more flexible interaction.

3. **Pause/Resume Timer Behavior:**  
  Currently, the timer resets on mouse hover and restarts on mouse leave. It would be beneficial to introduce a prop that can control whether to pause/resume or reset/restart the timer.

### Examples:

[Nuxt UI Notification Timeout with Progress Bar](https://ui.nuxt.com/components/notification#timeout)

[Nuxt UI Notification Actions](https://ui.nuxt.com/components/notification#actions)

### Proposed Props:

```ts
showProgressBar: {
  type: Boolean,
  default: false,
},
actions: {
  type: definePropType<
    { label: string, execute: () => void }[]
  >(Array),
},
timerControls: {
  type: String,
  values: ['pause-resume', 'reset-restart'],
  default: 'reset-restart',
}
```
*Default values prevent code modification for previous behavior after library update. However, this can be discussed.*

### Additional Suggestions:

1. **Keyboard Event Handler Documentation:**  
  The documentation currently lacks details on how the keyboard listener for delete, backspace and escape interacts with the timer. If `timerControls` is set to 'pause-resume', the listener logic may require adjustments.

2. **Refactoring:**  
  The current component can be refactored by moving logic into composables to enhance organization and maintainability.

**Comment by geforseN** [comment]
Hi @warmthsea, should I make a PR? I will handle the code, docs, and tests, and it will be a non-breaking change.

**Reply by warmthsea** [reply]
Very good feature, welcome PR.
It may not be merged immediately, more people are needed to review, but you can also use [pkg.pr.new](https://element-plus.org/en-US/guide/installation.html#version) to install your PR latest code at any time.

![screenshots](https://github.com/user-attachments/assets/aa1932b3-d148-4390-ad90-a531915c64f9)
